frontend/controller/actions/group.js

```diff
@@ -464,47 +465,49 @@ export default (sbp('sbp/selectors/register', {
           if (!state) return
 
           // We have indeed been removed
-          if (state.profiles?.[username]?.status === PROFILE_STATUS.REMOVED) {
+          if (state.profiles?.[userID]?.status === PROFILE_STATUS.REMOVED) {
+            console.warn('Requested to join group but we\'ve been removed. contractID=' + params.contractID)
+
             // We can't await as remove is using the same queue
             console.info('[gi.actions/group/join] We appear to have left the group. Removing the contract.', {
               contractID: params.contractID,
-              username,
               userID,
               hasSecretKeys,
               pendingKeyShares,
-              status: state.profiles?.[username]?.status
+              status: state.profiles?.[userID]?.status
             })
             sbp('chelonia/contract/remove', params.contractID).catch((e) => {
               console.error('[gi.actions/group/join] An error occurred while trying to remove the group contract', e)
             })
           } else {
-            console.warn('[gi.actions/group/join] Invalid or inconsistent state. We would appear to have been removed but aren\'nt', {
+            console.warn('[gi.actions/group/join] Invalid or inconsistent state. We would appear to have been removed but aren\'t', {
               contractID: params.contractID,
-              username,
               userID,
               hasSecretKeys,
               pendingKeyShares,
-              status: state.profiles?.[username]?.status
+              status: state.profiles?.[userID]?.status
             })
           }
         }).catch((e) => {
           console.error('[gi.actions/group/join] Error while completing the group removal process', {
             contractID: params.contractID,
-            username,
             userID,
             hasSecretKeys,
             pendingKeyShares,
-            status: state.profiles?.[username]?.status
+            status: state.profiles?.[userID]?.status
           }, e)
         })
       } else if (pendingKeyShares) {
         console.info('Requested to join group but already waiting for OP_KEY_SHARE. contractID=' + params.contractID)
+        sbp('chelonia/contract/cancelRemove', params.contractID)
       } else {
         console.warn('Requested to join group but the state appears invalid. contractID=' + params.contractID, { sendKeyRequest, hasSecretKeys, pendingKeyShares })
       }
     } catch (e) {
       console.error('gi.actions/group/join failed!', e)
       throw new GIErrorUIRuntimeError(L('Failed to join the group: {codeError}', { codeError: e.message }))
+    } finally {
+      await sbp('chelonia/contract/remove', params.contractID, { removeIfPending: true })
```

Comment:
What is this, why is it here, please add a comment explaining. This seems like a significant change with no explanation.

Reply:
Added comment